pkg/metallb/metallb_test.go

```diff
@@ -1 +1,405 @@
 package metallb
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/metallb/mlbtypes"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	metalLbGVK = schema.GroupVersionKind{
+		Group:   APIGroup,
+		Version: APIVersion,
+		Kind:    metalLb,
+	}
+	defaultMetalLbName         = "metallbio"
+	defaultMetalLbNsName       = "test-namespace"
+	defaultMetalLbNodeSelector = map[string]string{"test": "test"}
+)
+
+func TestMetalLbPull(t *testing.T) {
+	generateMetalLb := func(name, namespace string) *mlbtypes.MetalLB {
+		return &mlbtypes.MetalLB{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+			Spec: mlbtypes.MetalLBSpec{},
+		}
+	}
+
+	testCases := []struct {
+		name                string
+		namespace           string
+		addToRuntimeObjects bool
+		expectedError       error
+		client              bool
+	}{
+		{
+			name:                "metallbio",
+			namespace:           "test-namespace",
+			addToRuntimeObjects: true,
+			expectedError:       nil,
+			client:              true,
+		},
+		{
+			name:                "",
+			namespace:           "test-namespace",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("metallb 'name' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                "metallbio",
+			namespace:           "",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("metallb 'nsname' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                "metallbio",
+			namespace:           "test-namespace",
+			addToRuntimeObjects: false,
+			expectedError:       fmt.Errorf("metallb object metallbio doesn't exist in namespace test-namespace"),
+			client:              true,
+		},
+		{
```

Comment:
I wonder about the requirement of this check.
We don't have it in all unit tests.

Reply:
Can you please elaborate?